presto-hive/src/main/java/com/facebook/presto/hive/ColumnarBinaryHiveRecordCursor.java

```diff
@@ -636,6 +641,9 @@ private void parseColumn(int column)
         else if (BIGINT.equals(type)) {
             parseLongColumn(column);
         }
+        else if (INTEGER.equals(type)) {
+            parseLongColumn(column);
```

Comment:
it looks like there's no checking in `parseLongColumn` that a `HIVE_INT` is 4 bytes, so add a check there since that would be corrupt data